Tasks/Maven/maven2.js

```diff
@@ -198,12 +197,16 @@ mvnv.exec()
 .fail(function (err) {
     console.error(err.message);
     console.error("SonarQube analysis failed");
-    tl.exit(1)
+    sqRunFailed = true;
 })
-.then(function (code) {
```

Comment:
The root cause of the bug was here. If SQ was not enabled, the previous .then block would return and the code variable would be undefined.